File: src/main/java/org/acme/models/Book.java
package org.acme.models;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

public class Book implements Serializable {
	@Serial
	private static final long serialVersionUID = 1L;

	private String id;
	private String name;
	private String description;
	private LocalDateTime createdAt = LocalDateTime.now();
	private LocalDateTime updatedAt = LocalDateTime.now();

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public LocalDateTime getCreatedAt() {
		return createdAt;
	}

	public void setCreatedAt(LocalDateTime createdAt) {
		this.createdAt = createdAt;
	}

	public LocalDateTime getUpdatedAt() {
		return updatedAt;
	}

	public void setUpdatedAt(LocalDateTime updatedAt) {
		this.updatedAt = updatedAt;
	}
}

File: src/main/kotlin/org/acme/models/Book.kt
package org.acme.models

import org.acme.enums.BookGenre
import java.time.LocalDateTime

data class Book(
    var id: String? = null,
    var name: String = "",
    var description: String = "",
    var genre: BookGenre = BookGenre.NOT_SPECIFIED,
    val createdAt: LocalDateTime,
    var updatedAt: LocalDateTime,
) {
    constructor() : this(
        id = null,
        name = "",
        description = "",
        genre = BookGenre.NOT_SPECIFIED,
        createdAt = LocalDateTime.now(),
        updatedAt = LocalDateTime.now()
    )
}